dotCMS/src/main/java/com/dotcms/content/business/ContentletJsonAPIImpl.java

```diff
@@ -265,12 +310,27 @@ ImmutableContentlet toImmutable(
                     }
                 }
             }
+            //We're returning Tags as a comma separated string for backwards compatibility
+            //This is expected to be removed in the near future
+            if(field instanceof TagField && value instanceof Collection){
+                final Collection<String>tags = (Collection)value;
+                value = String.join(StringPool.COMMA,tags);
+            }
             map.put(field.variable(), value);
         }
 
         return map;
     }
 
+    /**
+     * These basically tells what system fields are accepted in the generate contentlet-json
+     * @param field
+     * @return
+     */
+    private boolean isAllowedSystemField(final Field field){
```

Comment:
I am wondering if this should be on the ContentTypeAPI

Reply:
I don't think so, it's private logic